nemo/collections/cv/models/visual_ctc_models.py

```diff
@@ -0,0 +1,694 @@
+        """
+        Sets up the test data loader via a Dict-like object.
+
+        Args:
+            test_data_config: A config that contains the information regarding construction
+                of an ASR Training dataset.
+
+        Supported Datasets:
+            -   :class:`~nemo.collections.cv.data.video_to_text.VideoToCharDataset`
+            -   :class:`~nemo.collections.asr.data.video_to_text.VideoToBPEDataset`
+            -   :class:`~nemo.collections.asr.data.video_to_text.TarredVideoToBPEDataset`
+        """
+        if 'shuffle' not in test_data_config:
+            test_data_config['shuffle'] = False
+
+        # preserve config
+        self._update_dataset_config(dataset_name='test', config=test_data_config)
+
+        self._test_dl = self._setup_dataloader_from_config(config=test_data_config)
+
+    @property
+    def input_types(self) -> Optional[Dict[str, NeuralType]]:
+        return {
+            "input_video_signal": NeuralType(('B', 'C', 'T', 'H', 'W'), VideoSignal(), optional=True),
+            "input_video_signal_length": NeuralType(tuple('B'), LengthsType(), optional=True),
+            "sample_id": NeuralType(tuple('B'), LengthsType(), optional=True),
+        }
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        return {
+            "outputs": NeuralType(('B', 'T', 'D'), LogprobsType()),
+            "encoded_lengths": NeuralType(tuple('B'), LengthsType()),
+            "greedy_predictions": NeuralType(('B', 'T'), LabelsType()),
+        }
+
+    @typecheck()
+    def forward(self, input_video_signal=None, input_video_signal_length=None):
+        """
+        Forward pass of the model.
+
+        Args:
+            input_video_signal: Tensor that represents a batch of video signals,
+                of shape [B, T, H, W, C]. T here represents timesteps, H height, W width and C channels
+            input_video_signal_length: Vector of length B, that contains the individual lengths of the video
+                sequences.
+
+        Returns:
+            A tuple of 3 elements -
+            1) The log probabilities tensor of shape [B, T, D].
+            2) The lengths of the acoustic sequence after propagation through the encoder, of shape [B].
+            3) The greedy token predictions of the model of shape [B, T] (via argmax)
+        """
+
+        # Preprocessing
+        processed_video_signal, processed_video_signal_length = self.video_preprocessor(
+            input_signal=input_video_signal, length=input_video_signal_length
+        )
+
+        # Augmentation
+        processed_video_signal = self.video_augmentation(
+            input_signal=processed_video_signal, length=processed_video_signal_length
+        )
+
+        # Front-end Networks
+        processed_video_signal, processed_video_signal_length = (
+            self.video_front_end(input_signal=processed_video_signal),
```

Comment:
Hmm for future proofing, can you make the front end take as input the processed_video_signal_length and return it as well? If it doesn't use it, its essentially a no-op but useful for future. We had issues where spec aug initially ignored seq len but then we found its useful and had to redo a whole bunch of classes.

Reply:
Sure, we can imagine a front-end that modify the signal length in the future

Reply:
done